lib/service/service.go

```diff
@@ -1301,28 +1332,29 @@ func NewTeleport(cfg *servicecfg.Config) (*TeleportProcess, error) {
 				process.logger.InfoContext(process.ExitContext(), "Exported autoupdates endpoint.", "addr", resolverAddr.String())
 				return nil
 			})
-		}
-
-		driver, err := uw.NewDriver(upgraderKind)
-		if err != nil {
-			return nil, trace.Wrap(err)
-		}
+			fallthrough
```

Comment:
I'm a bit concerned about fallthroughs in switches as big/complex as this. Fallthroughs are not common and it's too easy to change a clause without realizing it affects the other, especially if it lacks test coverage (which I'm assuming we do?).

Maybe refactor the repeated logic into a function and call it in both places?

Reply:
Removed fallthrough